Dockerfile.assisted-test-infra

```diff
@@ -39,33 +48,20 @@ RUN dnf -y install --enablerepo=crb \
 # running with a random user.
 RUN git config --system --add safe.directory '*'
 
-RUN curl --retry 5 --connect-timeout 30 -Lo packer.zip https://releases.hashicorp.com/packer/1.9.4/packer_1.9.4_linux_386.zip && unzip packer.zip -d /usr/bin/ && mv /usr/bin/packer /usr/bin/packer.io && rm -rf packer.zip
-RUN curl --retry 5 --connect-timeout 30 -Lo terraform.zip https://releases.hashicorp.com/terraform/1.6.3/terraform_1.6.3_linux_amd64.zip && unzip terraform.zip -d /usr/bin/ && rm -rf terraform.zip
-RUN curl --retry 5 --connect-timeout 30 -L https://github.com/containers/podman/releases/download/v3.4.4/podman-remote-static.tar.gz -o "/tmp/podman-remote3.tar.gz" && \
-    tar -zxvf /tmp/podman-remote3.tar.gz && \
-    mv podman-remote-static /tools/podman-remote3 && \
-    rm -f /tmp/podman-remote3.tar.gz &&\
-    curl --retry 5 --connect-timeout 30 -L https://github.com/containers/podman/releases/download/v4.1.1/podman-remote-static.tar.gz -o "/tmp/podman-remote4.tar.gz" && \
-    tar -zxvf /tmp/podman-remote4.tar.gz && \
-    mv podman-remote-static /tools/podman-remote4 && \
-    rm -f /tmp/podman-remote4.tar.gz
+RUN mkdir /tools --mode g+xw; \
+    curl ${CURL_OPT} -Ls https://github.com/containers/podman/releases/download/v4.4.4/podman-remote-static-linux_amd64.tar.gz | tar zxvf - -O > /tools/podman-remote4; \
+    curl ${CURL_OPT} -Ls https://github.com/containers/podman/releases/download/v3.1.1/podman-remote-static.tar.gz | tar zxvf - -O > /tools/podman-remote3; \
+    curl ${CURL_OPT} -Ls https://releases.hashicorp.com/packer/${PACKER_VERSION}/packer_${PACKER_VERSION}_linux_amd64.zip | zcat > /usr/bin/packer.io; \
+    curl ${CURL_OPT} -Ls https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip | zcat >  /usr/bin/terraform; \
+    curl ${CURL_OPT} -Ls https://go.dev/dl/go${GO_VERSION}.linux-amd64.tar.gz| tar -zxvf -  -C /usr/local/ go/bin/*; \
+    chmod +x  /tools/podman-remote* /usr/bin/packer.io /usr/bin/terraform /usr/local/go/bin/* 
 
 COPY --from=quay.io/ocp-splat/govc:v0.29.0 /govc /usr/local/bin
+COPY . . 
 
-WORKDIR /home/assisted-test-infra
-
-COPY requirements.txt requirements-dev.txt ./
-COPY --from=service /clients/assisted_service_client-*.whl /build/pip/
 RUN pip3 install --upgrade pip && \
-      pip3 install --no-cache-dir -I -r ./requirements.txt -r ./requirements-dev.txt && \
-      pip3 install --upgrade /build/pip/*
-
-RUN curl --retry 5 --connect-timeout 30 -s https://storage.googleapis.com/golang/go1.18.1.linux-amd64.tar.gz | tar -C /usr/local -xz
-ENV GOPATH=/go
-ENV GOCACHE=/go/.cache
-ENV PATH=$PATH:/usr/local/go/bin:/go/bin
-
-COPY . .
+		pip3 install --no-cache-dir  assisted-service-client && \
+    pip3 install --no-cache-dir -I -r ./requirements.txt -r ./requirements-dev.txt 
```

Comment:
This was before copy all so that we could leverage caching (no need to pip install if any file changes apart from requirements)